netaddr.go

```diff
@@ -737,3 +702,16 @@ func (p *IPPrefix) UnmarshalText(text []byte) error {
 func (p IPPrefix) String() string {
 	return fmt.Sprintf("%s/%d", p.IP, p.Bits)
 }
+
+// mask preserves the first bits bits of b, and zeroes the
+// rest. Assumes that bits is at most the total number of bits in b.
+func mask(b []byte, bits uint8) {
+	skip, partial := int(bits/8), bits%8
+	if partial != 0 {
+		b[skip] = b[skip] & ^byte(0xff >> partial)
+		skip++
+	}
+	for i := skip; i < len(b); i++ {
```

Comment:
you might get a better compiler result by:

```
   b := b[skip:]
   for i := range b {
        b[i] = 0
   }
```
... then you get a memclr.

Reply:
(the compiler optimizes that zeroing pattern)

Reply:
Results: better for IPv6 when clearing most of the addr, same or worse in other cases. Seems like memclr has a high setup cost, so for a few bytes it's better to just eat the inline clearing cost.

Looking in godbolt (https://go.godbolt.org/z/k5MFn_), your code does indeed result in a memclr call, whereas mine is icky loop iteration with bounds checking. But the memclr doesn't get inlined, so it seems to end up worse in terms of basic cost (function call vs. speculable loop) ?

WDYT?

```
IPPrefixMasking/IPv4_/32-8          27.5ns ± 3%    30.4ns ± 2%  +10.39%  (p=0.008 n=5+5)
IPPrefixMasking/IPv4_/17-8          30.9ns ± 2%    35.8ns ± 2%  +15.71%  (p=0.008 n=5+5)
IPPrefixMasking/IPv4_/0-8           32.5ns ± 3%    35.2ns ± 1%   +8.50%  (p=0.008 n=5+5)
IPPrefixMasking/IPv6_/128-8         44.3ns ± 8%    43.9ns ± 3%     ~     (p=0.968 n=5+5)
IPPrefixMasking/IPv6_/65-8          47.7ns ± 5%    47.6ns ± 2%     ~     (p=0.802 n=5+5)
IPPrefixMasking/IPv6_/0-8           53.2ns ± 1%    46.3ns ± 3%  -13.08%  (p=0.008 n=5+5)
IPPrefixMasking/IPv6_zone_/128-8    77.3ns ± 2%    87.4ns ± 2%  +13.07%  (p=0.008 n=5+5)
IPPrefixMasking/IPv6_zone_/65-8     94.8ns ± 3%    86.1ns ± 2%   -9.26%  (p=0.008 n=5+5)
IPPrefixMasking/IPv6_zone_/0-8      98.5ns ± 1%    86.6ns ± 2%  -12.04%  (p=0.008 n=5+5)
```